### PR TITLE
Immutable version (suggested new pre-release minor 0.4)

### PR DESCRIPTION
**Pull request by richard-ejem**
Re-written code from #10 as immutable only version to new branch

### CODE REVIEW

**Comment by Majkl578** [comment]
Can you avoid the changes to whitespaces/eol etc.? It makes the diff really harder to read... :/

File: src/Kdyby/Clock/IDateTimeProvider.php [hunk 1]
```diff
@@ -5,20 +5,20 @@
...
 
-
 /**
  * @author Michael Moravec
  */
 interface IDateTimeProvider
 {
 	/**
-	 * @return \DateTime|\DateTimeInterface
+	 * @return \DateTimeInterface
```
Comment (Majkl578) [inline src/Kdyby/Clock/IDateTimeProvider.php]: These methods should probably return \DateTimeImmutable directly.

Comment (richard-ejem) [inline src/Kdyby/Clock/IDateTimeProvider.php]: fixed

**Comment by fprochazka** [comment]
@richard-ejem thank you!

**Pull request closed without merging**